presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -90,6 +91,84 @@ public void testDecimal()
         try (ConnectedStatement connectedStatement = newStatement()) {
             checkRepresentation(connectedStatement.getStatement(), "0.1", Types.DECIMAL, new BigDecimal("0.1"));
             checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL, new BigDecimal("0.1"));
+            checkRepresentation(connectedStatement.getStatement(), "0.12", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 0.12D));
+            checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 0.1D));
+            checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getLong(column), 0));
+
+            long outsideOfDoubleExactRange = 9223372036854775774L;
+            //noinspection ConstantConditions
+            Verify.verify((long) (double) outsideOfDoubleExactRange - outsideOfDoubleExactRange != 0, "outsideOfDoubleExactRange should not be exact-representable as a double");
+            checkRepresentation(connectedStatement.getStatement(), String.format("DECIMAL '%s'",
+                    outsideOfDoubleExactRange), Types.DECIMAL,
+                    (rs, column) -> {
+                        assertEquals(rs.getObject(column), new BigDecimal("9223372036854775774"));
+                        assertEquals(rs.getBigDecimal(column), new BigDecimal("9223372036854775774"));
+                        assertEquals(rs.getLong(column), 9223372036854775774L);
+                        assertEquals(rs.getDouble(column), 9.223372036854776E18);
+                        assertEquals(rs.getString(column), "9223372036854775774");
+                    });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR ''", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getBigDecimal(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: ");
+            });
+
+            checkRepresentation(connectedStatement.getStatement(), "VARCHAR '123a'", Types.VARCHAR, (rs, column) -> {
+                assertThatThrownBy(() -> rs.getBigDecimal(column))
+                        .isInstanceOf(SQLException.class)
+                        .hasMessage("Value is not a number: 123a");
+            });
```

Comment:
I just though a new test case is due `VARCHAR '123e-1'` (can return big decimal with value 12.3)

Reply:
fixed